core/templates/dev/head/services/IdGenerationServiceSpec.js

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for IdGenerationService.
+ */
+
+describe('IdGenerationService', function() {
+  var IdGenerationService = null;
+
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    IdGenerationService = $injector.get('IdGenerationService');
+  }));
+
+  it('should generate a random id of fixed length', function() {
+    expect(IdGenerationService.generateNewId()).toMatch(/^[a-z 0-9]{10}$/);
```

Comment:
Why do you need the space in the regex? Wouldn't a-z and 0-9 be the full charset?

Reply:
@seanlip sorry that was a mistake. Made the changes